bundles/org.openhab.binding.intellicenter2/src/main/java/org/openhab/binding/intellicenter2/internal/handler/IntelliCenter2ThingHandler.java

```diff
@@ -0,0 +1,157 @@
+                    return;
+                }
+                final T model = queryModel(protocol);
+                getProtocol().subscribe(this, model.asRequestObject());
+                updateStatus(ThingStatus.ONLINE);
+                updateState(model);
+            } catch (Exception e) {
+                logger.error("Error refreshing model", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        scheduler.execute(() -> {
+            if (getProtocol() != null) {
+                getProtocol().unsubscribe(this);
+            }
+            updateStatus(ThingStatus.OFFLINE);
+        });
+    }
+
+    @Nullable
+    protected ICProtocol getProtocol() {
+        if (getBridgeHandler() != null) {
+            return getBridgeHandler().getProtocol();
+        }
+        return null;
+    }
+
+    @Nullable
+    protected IntelliCenter2BridgeHandler getBridgeHandler() {
+        if (getBridge() == null) {
+            return null;
+        }
+        return (IntelliCenter2BridgeHandler) getBridge().getHandler();
+    }
+
+    @Override
+    public void handleCommand(@NonNull ChannelUID channelUID, @NonNull Command command) {
+        logger.debug("Handling {} for {}", channelUID, command);
+        final ICProtocol protocol = getProtocol();
+        if (protocol == null) {
+            logger.error("handleCommand had a null protocol for {} {}", channelUID, command);
+            return;
+        }
+        if (command instanceof RefreshType) {
+            final T model = queryModel(protocol);
+            logger.debug("updateState uid: {}, id: {}, model: {}", channelUID, channelUID.getId(), model);
+            updateState(channelUID, model);
+        }
+    }
+
+    @Override
+    public void onNotifyList(ResponseObject response) {
+        try {
+            if (!isInitialized()) {
+                logger.warn("Notified of a NotifyList response but this is not initialized: {}", response);
+                return;
+            }
+            final T model = createFromResponse(response);
+            logger.debug("onNotifyList {}", model);
+            response.getParams().forEach((k, v) -> {
+                var id = toChannelId(k);
+                if (id != null) {
+                    var channel = getThing().getChannel(id);
+                    if (channel != null) {
+                        updateState(channel.getUID(), model);
+                    } else {
+                        logger.warn("Unable to find a channel to update for {}", id);
+                    }
+                } else {
+                    logger.warn("Received notifyList update for attribute {} that doesn't map to a channelId", k);
+                }
+            });
+        } catch (Exception e) {
+            logger.error("Error handling onNotifyList", e);
```

Comment:
Please reduce log level unless this is a severe error that will prevent the binding from working correctly.